context/src/main/java/io/opentelemetry/context/propagation/TextMapGetter.java

```diff
@@ -33,4 +35,20 @@ public interface TextMapGetter<C> {
    */
   @Nullable
   String get(@Nullable C carrier, String key);
+
+  /**
+   * If implemented, returns all values for a given {@code key} in order, or returns an empty list.
+   *
+   * <p>The default method returns the first value of the given propagation {@code key} as a
+   * singleton list, or returns an empty list.
+   *
+   * @param carrier carrier of propagation fields, such as an http request.
+   * @param key the key of the field.
+   * @return the first value of the given propagation {@code key} as a singleton list, or returns an
+   *     empty list.
+   */
+  default List<String> getList(@Nullable C carrier, String key) {
```

Comment:
It might be necessary for us to add this in an experimental fashion, rather than jumping straight to the stable API.

For example, we could have an `ExtendedTextMapGetter` interface in `opentelemetry-api-incubator` or in an internal package of `opentelemetry-context`. Then, `W3CBaggagePropagator` could check `if (getter instanceOf ExtendedTextMapGetter)` and adjust its behavior accordingly.

If we do this, I think it would have to go in `opentelemetry-context` instead of `opentelemetry-api-incubator` since `W3CBaggagePropagator` doesn't have access to `opentelemetry-api-incubator`.

Reply:
Makes sense to me

Reply:
Ok, this is done - please take a look